Multiple definition errors for Barkeep functions when included in multiple .cpp files

Hi,

Thanks for the amazing progress bar library!

I’m using Barkeep in a C++ project with multiple source files. I include barkeep.h in multiple .cpp files, as each file uses its own progress bar. However, I get the following compilation error:

for example:
```shell
CMakeFiles/SpinWalk.dir/src/sim/monte_carlo.cu.o: in function `barkeep::as_duration(std::variant<std::chrono::duration<double, std::ratio<1l, 1l> >, double>)':
tmpxft_0009d674_00000000-6_monte_carlo.compute_89.cudafe1.cpp:(.text+0x399): multiple definition of `barkeep::as_duration(std::variant<std::chrono::duration<double, std::ratio<1l, 1l> >, double>)'; CMakeFiles/SpinWalk.dir/src/phantom/phantom_sphere.cpp.o:phantom_sphere.cpp:(.text+0xb): first defined here
``` 
If I include Barkeep in only one .cpp file, the issue does not occur.
There are multiple errors like this for other Barkeep functions as well.
There’s no recursive inclusion of headers.

I suggest mark the functions as `inline`.

--- a/barkeep/barkeep.h
+++ b/barkeep/barkeep.h
@@ -365,7 +365,7 @@ struct AnimationConfig {
   bool show = true; ///< show the animation immediately after construction
 };
 
-Duration as_duration(std::variant<Duration, double> interval) {
+inline Duration as_duration(std::variant<Duration, double> interval) {
   if (std::holds_alternative<Duration>(interval)) {
     return std::get<Duration>(interval);
   } else {
@@ -421,7 +421,7 @@ class AnimationDisplay : public BaseDisplay {
 
 /// Convenience factory function to create a shared_ptr to AnimationDisplay.
 /// Prefer this to constructing AnimationDisplay directly.
-auto Animation(const AnimationConfig& cfg = {}) {
+inline auto Animation(const AnimationConfig& cfg = {}) {
   return std::make_shared<AnimationDisplay>(cfg);
 }
 
@@ -468,7 +468,7 @@ class StatusDisplay : public AnimationDisplay {
 
 /// Convenience factory function to create a shared_ptr to StatusDisplay.
 /// Prefer this to constructing StatusDisplay directly.
-auto Status(const AnimationConfig& cfg = {}) {
+inline auto Status(const AnimationConfig& cfg = {}) {
   return std::make_shared<StatusDisplay>(cfg);
 }
 
@@ -1032,13 +1032,13 @@ class CompositeDisplay : public BaseDisplay {
 
 /// Convenience factory function to create a shared_ptr to CompositeDisplay.
 /// Prefer this to constructing CompositeDisplay directly.
-auto Composite(const std::vector<std::shared_ptr<BaseDisplay>>& displays,
+inline auto Composite(const std::vector<std::shared_ptr<BaseDisplay>>& displays,
                std::string delim = " ") {
   return std::make_shared<CompositeDisplay>(displays, std::move(delim));
 }
 
 /// Pipe operator can be used to combine two displays into a Composite.
-auto operator|(std::shared_ptr<BaseDisplay> left,
+inline auto operator|(std::shared_ptr<BaseDisplay> left,
                std::shared_ptr<BaseDisplay> right) {
   return std::make_shared<CompositeDisplay>(
       std::vector{std::move(left), std::move(right)});

